plugin/src/main/java/io/jenkins/plugins/remotingkafka/KubernetesFactoryAdapter.java

```diff
@@ -0,0 +1,170 @@
+
+        ConfigBuilder builder;
+        // configure from kubeconfig
+        if (credentials instanceof FileCredentials) {
+            LOGGER.log(FINE, "Configuring Kubernetes client from kubeconfig file");
+            try (InputStream is = ((FileCredentials) credentials).getContent()) {
+                Config config = Config.fromKubeconfig(IOUtils.toString(is, StandardCharsets.UTF_8));
+                builder = new ConfigBuilder(config);
+            }
+        } else {
+            // autoconfigure if url is not set
+            if (StringUtils.isBlank(serviceAddress)) {
+                LOGGER.log(FINE, "Autoconfiguring Kubernetes client");
+                builder = new ConfigBuilder(Config.autoConfigure(null));
+            } else {
+                builder = new ConfigBuilder().withMasterUrl(serviceAddress);
+            }
+        }
+
+        if (credentials instanceof FileCredentials) {
+            // already handled above
+        } else if (credentials instanceof StringCredentials) {
+            final String token = ((StringCredentials) credentials).getSecret().getPlainText();
+            builder.withOauthToken(token);
+        } else if (credentials instanceof TokenProducer) {
+            final String token = ((TokenProducer) credentials).getToken(serviceAddress, caCertData, skipTlsVerify);
+            builder.withOauthToken(token);
+        } else if (credentials instanceof UsernamePasswordCredentials) {
+            UsernamePasswordCredentials usernamePassword = (UsernamePasswordCredentials) credentials;
+            builder.withUsername(usernamePassword.getUsername())
+                    .withPassword(Secret.toString(usernamePassword.getPassword()));
+        } else if (credentials instanceof StandardCertificateCredentials) {
+            StandardCertificateCredentials certificateCredentials = (StandardCertificateCredentials) credentials;
+            KeyStore keyStore = certificateCredentials.getKeyStore();
+            String alias = keyStore.aliases().nextElement();
+            X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+            Key key = keyStore.getKey(alias, Secret.toString(certificateCredentials.getPassword()).toCharArray());
+            builder.withClientCertData(Base64.encodeBase64String(certificate.getEncoded()))
+                    .withClientKeyData(pemEncodeKey(key))
+                    .withClientKeyPassphrase(Secret.toString(certificateCredentials.getPassword()));
+        }
+
+        if (skipTlsVerify) {
+            builder.withTrustCerts(true);
+        }
+
+        if (caCertData != null) {
+            builder.withCaCertData(Base64.encodeBase64String(caCertData.getBytes(UTF_8)));
+        }
+
+        builder = builder.withRequestTimeout(readTimeout * 1000).withConnectionTimeout(connectTimeout * 1000);
+        builder.withMaxConcurrentRequestsPerHost(maxRequestsPerHost);
+
+        if (!StringUtils.isBlank(namespace)) {
```

Comment:
Generally more readable to not have negative if statements. I'd see if it isn't too messy to make this positive case.